tests/integration/ha_tests/helpers.py

```diff
@@ -79,6 +79,9 @@ async def are_all_db_processes_down(ops_test: OpsTest, process: str) -> bool:
 
                     # If something was returned, there is a running process.
                     if len(processes) > 0:
+                        logger.info("Unit %s not yet down" % unit.name)
+                        # Try to rekill the unit
+                        await send_signal_to_process(ops_test, unit.name, process, signal)
```

Comment:
Observing the log, there's usually at least one unit stuck. I guess that it manages to escape systemd's restart condition/Patroni's loop_wait and gets revived, so killing it again to make sure.